### ISSUE TITLE
Implementing $limit $sort etc...

### ISSUE THREAD

**Issue by mrSidX** [steineey/node-red-contrib-mongodb4#16]
I'm having trouble getting $limit and $sort options to work with this node.

Is there an example that demonstrates this?

**Comment by mrSidX** [comment]
I was able to get an aggregate operation going by doing the double brackets on the array.
As mentioned before by another, though I was using Find and doing something similar to a different node.
The confusing part was the requirement of the double brackets, and may be factor by lack of my understanding of the core functionality.

**Comment by steineey** [comment]
Hi mrSidX, 
sry for the late reply. You can use the find operation with $sort, $limit and also $projection. The `msg.payload` array is interpreted as an argument list for the `collection.find(query, options)` method of the NodeJS Mongodb driver.

Here's an example, how to do this in a function node:
```js
// find query
const query = {};
// find options
const options = {
    sort: {age: 1}, 
    projection: {name: 1},
    limit: 10
};
// payload for mongodb4 node
msg.payload = [query, options];
return msg;
```
[Here is a list of all the find options](https://mongodb.github.io/node-mongodb-native/4.12/interfaces/FindOptions.html).

Another method would be an aggregation. The aggregation function also supports an option argument.
```js
// aggregate pipeline
const pipeline = [{
    $sort:{age: 1}
}, {
    $project: {
        name: 1
    }
},{
    $limit: 10
}];
// aggregate options
const options = {
    allowDiskUse: true
};
// payload for mongodb4 node
msg.payload = [pipeline, options];
return msg;
```
I know that this might be confusing, if you have just a pipeline array and you have to wrap it inside another array. But I don't have a better solution to pass multiple argument to the mongodb driver.
For me it was important to design this node in a way that you can use all the functions of the native MongoDB driver without any restrictions.

**Comment by mrSidX** [comment]
I see.  I was using the **"$"** to prefix the values in the **_options_** object, and was getting errors returned something about the top level keys.
I tried all sorts as structured similar to your **find** example, and nothing, and I think I tried without the $, but it was a long day, so who knows.  I'm sure I'll get around to testing further in  development.

Meanwhile the Aggregation operation works a charm, and with large payload returns, I get the data back much quicker.

Thanks for clarifying the code examples for us.  I'll hope to report back my test on the **find** operation shortly.
